### PR TITLE
ROX-24001: Check details pagination

### PR DESCRIPTION
**Pull request by bradr5**
## Description

Adds pagination to the Compliance V2 Check Details table

## Checklist
- [ ] Investigated and inspected CI test results
- [ ] Unit test and regression tests added
- [ ] Evaluated and added CHANGELOG entry if required
- [ ] Determined and documented upgrade steps
- [ ] Documented user facing changes (create PR based on [openshift/openshift-docs](https://github.com/openshift/openshift-docs) and merge into [rhacs-docs](https://github.com/openshift/openshift-docs/tree/rhacs-docs))

If any of these don't apply, please comment below.

## Testing Performed

Pagination:
Screenshot 2024-05-28 at 12 53 09 PM

Page 2:
Screenshot 2024-05-28 at 1 02 08 PM

### CODE REVIEW

File: ui/apps/platform/src/Containers/ComplianceEnhanced/Coverage/CheckDetailsPage.tsx [hunk 1]
```diff
@@ -31,42 +32,45 @@ function sortCheckStats(a: ComplianceCheckStatusCount, b: ComplianceCheckStatusC
 function CheckDetails() {
     const { checkName, profileName } = useParams();
     const [currentDatetime, setCurrentDatetime] = useState(new Date());
+    const pagination = useURLPagination(10);
```
Comment (pedrottimark) [inline ui/apps/platform/src/Containers/ComplianceEnhanced/Coverage/CheckDetailsPage.tsx:35]: What are pro and con of a constant for page size like this:
https://github.com/stackrox/stackrox/blob/master/ui/apps/platform/src/Containers/Vulnerabilities/constants.ts#L1

Maybe factor out to the level of a folder like Coverage or Schedules to balance consistency with flexibility?

File: ui/apps/platform/src/services/ComplianceResultsService.ts [hunk 2]
```diff
@@ -46,11 +46,19 @@ export type ListComplianceCheckClusterResponse = {
  */
 export function getComplianceProfileCheckResult(
     profileName: string,
-    checkName: string
+    checkName: string,
+    page: number,
+    perPage: number
```
Comment (pedrottimark) [inline ui/apps/platform/src/services/ComplianceResultsService.ts:51]: Here is a question for the mental slow cooker, not necessarily for this contribution.

As the number of arguments of a service function increases, what are pro and con:
* Two string arguments like `profileName` and `checkName` which correspond to order in endpoint address but two number arguments `page` and `perPage` for which order is more arbitrary (although consistent between many functions).
    Compare to following `getComplianceProfileResults(profileName, sortOption, page, perPage)` function.
* One auxiliary object argument whose property names correspond to PatternFly semantics.
    * `getComplianceProfileCheckResult(profileName, checkName, { page, perPage })`
    * `getComplianceProfileResults(profileName, { page, perPage, sortOption })`

```ts
export type Whatchamacallit = {
    page: number;
    perPage: number;
    sortOption?: ApiSortOption;
};
```

With a helper function that converts `offset: (page - 1) * perPage, limit: perPage` for request params:

https://github.com/stackrox/stackrox/blob/master/ui/apps/platform/src/services/types.ts#L3-L7

```ts
export type Pagination = {
    offset: number;
    limit: number;
    sortOption?: ApiSortOption;
};
```

**Review by pedrottimark — APPROVED** [review]
Another clear and good-sized chunk.

File: ui/apps/platform/src/Containers/ComplianceEnhanced/Coverage/CheckDetailsPage.tsx (hunk 1 above)
Comment (bradr5) [reply ui/apps/platform/src/Containers/ComplianceEnhanced/Coverage/CheckDetailsPage.tsx:35]: I think that's a good idea. I was planning on making that constants file for the sortable/searchable fields as well. Seems reasonable to include all of that in the same file.

File: ui/apps/platform/src/services/ComplianceResultsService.ts (hunk 2 above)
Comment (bradr5) [reply ui/apps/platform/src/services/ComplianceResultsService.ts:51]: Interesting thought. It has felt weird to me having the "critical" (profileName, checkName, etc) alongside the "non-critical" (pagination, searchField, etc) in the list of parameters. Especially when the "non-critical" that I just described seem to be included everywhere.

I'll dabble with the idea as I implement sorting on this table

**Pull request merged**